### PR TITLE
Update assertJ core to 3.18.1

### PR DESCRIPTION
**Pull request by wendigo**
Preparatory for https://github.com/trinodb/trino/pull/6627

### CODE REVIEW

**Comment by wendigo** [comment]
Iceberg failed due to https://github.com/trinodb/trino/issues/5758

File: core/trino-main/src/test/java/io/trino/sql/planner/planprinter/TestWindowOperatorStats.java [hunk 1]
```diff
@@ -21,18 +21,22 @@
 
 public class TestWindowOperatorStats
 {
+    // The assertion will fail if both actual and expected are primitive Double.NaN
+    // as Double.NaN != Double.NaN (as per Double#equals javadoc).
+    private static final Double NAN = Double.NaN;
```
Comment (findepi) [inline core/trino-main/src/test/java/io/trino/sql/planner/planprinter/TestWindowOperatorStats.java:26]: use `.isNaN()` instead

**Review by findepi — APPROVED** [review]

File: core/trino-main/src/test/java/io/trino/sql/planner/planprinter/TestWindowOperatorStats.java [hunk 2]
```diff
@@ -33,7 +33,7 @@ public void testEmptyDriverInfosList()
         assertThat(stats.getIndexCountPerDriverStdDev()).isNaN();
         assertThat(stats.getPartitionRowsStdDev()).isNaN();
         assertThat(stats.getRowsPerDriverStdDev()).isNaN();
-        assertThat(stats.getActiveDrivers()).isEqualTo(0);
-        assertThat(stats.getTotalDrivers()).isEqualTo(0);
+        assertThat(stats.getActiveDrivers()).isZero();
+        assertThat(stats.getTotalDrivers()).isZero();
```
Comment (findepi) [inline core/trino-main/src/test/java/io/trino/sql/planner/planprinter/TestWindowOperatorStats.java:37]: i prefer the previous version, but whatever.... they shouldn't have added `isZero` in the first place.

**Review by findepi — APPROVED** [review]

**Comment by findepi** [comment]
see red

**Comment by wendigo** [comment]
@findepi dependency download error

**Comment by wendigo** [comment]
I've dropped isZero @findepi

**Review by findepi — APPROVED** [review]

**Pull request merged**